catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -244,6 +245,19 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
         responseProps.put(SUGGESTION_RESULT_KEY, (Serializable) suggestionResults);
       }
 
+      if (solrResponse.getSpellCheckResponse() != null
+          && solrResponse.getResults().size() == 0
```

Comment:
❓ Does this mean we will only get misspelled results if there are no hits that match the current spelling?

Reply:
Yes that is correct.

Reply:
So if a single result contains the current spelling we won't get misspelled results?

Reply:
I'll use an example to make this clear. 
Suppose your database contains two metacards, one named `Normal`, the other named `Nromal`.
If you search for `Nromal`, no re-querying is done because you get a result (it's the exact same case as searching for `Normal`.
If you search for Norrrmalll, then you would re-query.

Reply:
This looks like a good solution for the first iteration of this functionality.

If we wanted to further refine the spellchecking mechanism in the future, I would imagine a solution along the lines of "After the user sends the query, results are returned along with the spelling suggestions to choose from. Selecting a spelling correction would re-query" or something 🤷‍♂️

Reply:
Yes that is correct. This is something that will be improved as other features fall in place.